htdocs/core/class/html.formfile.class.php

```diff
@@ -679,7 +679,7 @@ public function showdocuments($modulepart, $modulesubdir, $filedir, $urlsource,
 						$modulepart=$tmp[0];
 						$submodulepart=$tmp[1];
 					}
-			        $file=dol_buildpath('/'.$modulepart.'/core/modules/'.$modulepart.'/modules_'.$submodulepart.'.php', 0);
+			        $file=dol_buildpath('/'.$modulepart.'/core/modules/'.$submodulepart.'/modules_'.$submodulepart.'.php', 0);
```

Comment:
We should have only one dir into core/modules per module. And we can put several "modules_submodulepart" when we need several handler.
What make you think you need to change this too ?

Reply:
imagine you have a module call MyModule with 3 objects with documents : MyModule, ObjectA and ObjectB

/MyModule/core/modules/MyModule/modules_MyModule.php
/MyModule/core/modules/objecta/modules_objecta.php
/MyModule/core/modules/objectb/modules_objectb.php

Does I misunderstand  something ?

Reply:
If you have ObjectA and ObjectB, you should get
/MyModule/core/modules/mymodule/modules_objecta.php
/MyModule/core/modules/mymodule/modules_objectb.php
and not 
/MyModule/core/modules/objecta/modules_objecta.php
/MyModule/core/modules/objectb/modules_objectb.php

You can have a look at modulebuilder tree to have confirmation.

Reply:
Ok